### PR TITLE
 add file argument

### LINKED ISSUE #10: Add python script selection in argparse
with a `-f` / `--file` argparse argument, instead of blindly acting on all of them everytime.

This will be useful for #6

### PR DESCRIPTION
**Pull request by TheoMF**
fix #10

### CODE REVIEW

**Comment by nim65s** [comment]
This is now in conflict with the main branch. Could you rebase your commit and fix this ?

**Comment by TheoMF** [comment]
done

File: gepetuto/lint.py [hunk 1]
```diff
@@ -8,21 +8,22 @@
...
     """Lint python scripts."""
     LOG.info("linting tutorial sources.")
     for n in tp_id:
         LOG.debug(f"Looking for tp {n}")
-        lint_tp_folder(n)
+        folder = Path(f"tp{n}")
+        for python_file in folder.glob("*.py"):
+            lint_file(python_file)
+    for f in file:
+        lint_file(f)
```
Comment (nim65s) [inline gepetuto/lint.py:20]: The idea here is to filter files. Here you are adding new ones.
I guess we should iterate over `file` **or** glob for `*.py` when it is empty,

File: gepetuto/test.py [hunk 2]
```diff
@@ -18,6 +18,9 @@ def test(tp_id: List[int], **kwargs):
             LOG.debug(f"Checking {python_file}")
             check_call(["python", python_file])
             check_ipynb(n)
+    for f in file:
+        LOG.debug(f"Checking {f}")
+        check_call(["python", f])
```
Comment (nim65s) [inline gepetuto/test.py:23]: same here

**Review by nim65s — CHANGES_REQUESTED** [review]

File: gepetuto/lint.py (hunk 1 above)
Comment (TheoMF) [reply gepetuto/lint.py:20]: you mean that there may be file that aren't .py ones and we should filter them ?

Comment (nim65s) [reply gepetuto/lint.py:20]: I was not clear, sorry for that.
I mean that if we have the following structure:
```
├── 1_example_notebook.ipynb
├── 2_other_notebook.ipynb
├── tp1
│   ├── cholesky.py
│   └── example_script.py
└── tp2
    ├── cholesky.py
    └── something.py
```

- with no tp_id and no file, we should find all 4 python files
- with tp_id=1 and no file, we should get the 2 python files in tp1
- with no tp_id and file=tp1/cholesky.py, we should get 1 python file in tp1
- with tp_id=1 and file=tp1/cholesky.py, we should get 1 python file in tp1
- with tp_id=2 and file=tp1/cholesky.py, we should get 0 python files

We'll also have to add `-F` / `--filter` to allow:
- with no tp_id and no file and filter=cholesky, we should get 2 python files, one in each tp
But that's for a separate future PR I think.

Comment (TheoMF) [reply gepetuto/lint.py:20]: no problem, understood

File: gepetuto/lint.py [hunk 3]
```diff
@@ -8,21 +8,27 @@
...
     for n in tp_id:
         LOG.debug(f"Looking for tp {n}")
-        lint_tp_folder(n)
+        folder = Path(f"tp{n}")
+        for python_file in folder.glob("*.py"):
+            if file == []:
+                lint_file(python_file)
+            elif python_file in file:
+                lint_file(python_file)
+    if tp_id == []:
```
Comment (nim65s) [inline gepetuto/lint.py:23]: I don't see how this condition can be true. `tp_id` wil either be defined by the user to not the empty list, or not be defined by the user and auto-populated from `get_tp_id()`

**Review by nim65s — COMMENTED** [review]
This is not easy. I think we'll need to unit test this behaviour. For just the sake of this PR, we can merge it if you run manual tests with debug on a test folder and check that the logs correspond to what we want.

File: gepetuto/lint.py (hunk 3 above)
Comment (TheoMF) [reply gepetuto/lint.py:23]: if someone has no tp folder and doesn't specify tp_id it could happen but i agree it shouldn't, should I still remove it ?

Comment (nim65s) [reply gepetuto/lint.py:23]: Yes, I think the logic between this `if` and its previous block should be changed. Probably we don't want to glob on `*.py` if `-f` is provided.

File: .pre-commit-config.yaml [hunk 4]
```diff
@@ -7,6 +7,7 @@ repos:
   rev: v0.0.281
   hooks:
   - id: ruff
+    exclude: gepetuto/magic.py
```
Comment (nim65s) [inline .pre-commit-config.yaml:10]: This won't prevent manual runs of `ruff .` to fail. I think you should instead add hints in the file : either a `  # noqa: F821` in the end of each line where there is a `get_ipython`, or a `# ruff: noqa: F821` at the begining of the file.

A better solution might be to add a `def run(self, magic)` method, which run `get_ipython().run_line_magic(magic)` with the correct noqa only once, and use this method in the other places.

**Review by nim65s — APPROVED** [review]
This looks better, thanks !

**Pull request merged**